ratelimit/http.go

```diff
@@ -38,58 +38,54 @@ type middleware struct {
 
 // WithLocalRateLimiter returns a middleware that controls the amount of requests per tenant using an in-memory store.
 func WithLocalRateLimiter(configs ...Config) Middleware {
-	middlewares := make(map[string][]middleware)
+	var mWare middleware
 	for _, c := range configs {
-		middlewares[c.Tenant] = append(middlewares[c.Tenant], middleware{c.Matcher, httprate.NewRateLimiter(
+		mWare = middleware{c.Matcher, httprate.NewRateLimiter(
 			c.Limit,
 			c.Window,
 			nil,
-		).Handler})
+		).Handler}
 	}
 
-	return combine(middlewares)
+	return combine(mWare)
 }
 
 // WithSharedRateLimiter returns a middleware that controls the amount of requests per tenant using an external service.
 func WithSharedRateLimiter(logger log.Logger, client *Client, configs ...Config) Middleware {
 	logger = log.With(logger, "component", "rate limiter")
 
-	middlewares := make(map[string][]middleware)
+	var mWare middleware
 	for _, c := range configs {
-		middlewares[c.Tenant] = append(middlewares[c.Tenant],
-			middleware{c.Matcher, rateLimiter{logger, client, &request{
-				name:     requestName,
-				key:      fmt.Sprintf("%s:%s", c.Tenant, c.Matcher.String()),
-				limit:    int64(c.Limit),
-				duration: c.Window.Microseconds(),
-			}}.Handler})
+		mWare = middleware{c.Matcher, rateLimiter{logger, client, &request{
+			name:     requestName,
+			key:      fmt.Sprintf("%s:%s", c.Tenant, c.Matcher.String()),
+			limit:    int64(c.Limit),
+			duration: c.Window.Microseconds(),
+		}}.Handler}
 	}
 
-	return combine(middlewares)
+	return combine(mWare)
 }
 
-func combine(middlewares map[string][]middleware) func(next http.Handler) http.Handler {
+func combine(mWare middleware) func(next http.Handler) http.Handler {
```

Comment:
nit `mWare` is a strange naming here, suggest either typical golang `mw` or full name `middleware`.